File: app/src/main/java/com/smu/team_andeu/data/RoutineDao.java
package com.smu.team_andeu.data;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

@Dao
public interface RoutineDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(Routine routine);

    @Query("SELECT * FROM routine_table WHERE routineId = :rId")
    LiveData<Routine> getRoutine(int rId);
}

File: app/src/main/java/com/smu/team_andeu/data/RoutineDao.kt
package com.smu.team_andeu.data


import android.database.Cursor
import androidx.lifecycle.LiveData
import androidx.room.*

@Dao
interface RoutineDao {

    @Query("SELECT routineId as _id, r_name FROM routine WHERE category = :categoryId")
    fun loadRawRoutineByCategory(categoryId: Int): Cursor //LiveData<Cursor>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(routine: Routine?)

    @Query("SELECT * FROM routine WHERE routineId = :rId")
    fun getRoutine(rId: Int): LiveData<Routine>

    // 초기 넣기 : 비동기식 처리중
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertAll(routines: List<Routine>)

    @Update(entity = Routine::class)
    fun updateRutineMaxOrder(routine:RoutineOrder)
}